src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -310,19 +312,14 @@ private static IntPtr GetSslMethod(bool isServer, long options)
             return method;
         }
 
-        //TODO See if SSL_CTX_Set_quite_shutdown can be used
         private static void Disconnect(IntPtr sslPtr)
         {
             if (IntPtr.Zero != sslPtr)
             {
                 int retVal = libssl.SSL_shutdown(sslPtr);
-                if (0 == retVal)
+                if (retVal < 0)
                 {
-                    retVal = libssl.SSL_shutdown(sslPtr);
-                    if (retVal < 0)
-                    {
-                        libssl.SSL_get_error(sslPtr, retVal);
-                    }
+                    libssl.SSL_get_error(sslPtr, retVal);
```

Comment:
Does this work with an error queue like the rest of OpenSSL does?  If so, we wouldn't want to convert it to a Debug/Assert like that, as the SSL_get_error call would be removed in release and we'd leave the error in the queue.  We'd instead want to do something like:
```
if (retVal < 0)
{
    int err = libssl.SSL_get_error(ssPtr, retVal);
    Debug.Fail(...);
}
```